cpp/zotaut/validate_harvested_records.cc

```diff
@@ -466,13 +481,22 @@ int Main(int argc, char *argv[]) {
             LOG_INFO("Record " + record.getControlNumber() + " is valid.");
             valid_records_writer->write(record);
         } else {
-            const std::string error_messages(StringUtil::Join(reasons_for_being_invalid, "\n"));
-            LOG_WARNING("Record " + record.getControlNumber() + " is invalid:\n" + error_messages);
-            ++missed_expectation_count;
-            if (update_db_errors)
-                upload_tracker.archiveRecord(record, ZoteroHarvester::Util::UploadTracker::DeliveryState::ERROR,
-                                             error_messages);
-            delinquent_records_writer->write(record);
+            if (std::find(reasons_for_being_invalid.begin(), reasons_for_being_invalid.begin(), ONLINE_FIRST_OR_EARLY_VIEW_MESSAGE)
+                          != reasons_for_being_invalid.end())
+            {
+                online_first_records_writer->write(record);
+                ++online_first_record_count;
+                upload_tracker.archiveRecord(record, ZoteroHarvester::Util::UploadTracker::DeliveryState::ONLINE_FIRST,
```

Comment:
Please also have a close look at archiveRecord. Inside that function we should treat ONLINE_FIRST similar to the other DELIVERY_STATES_TO_RETRY states, so that the existing entry can be updated (and maybe set to DELIVERED)-

Reply:
I would intentionally locate the removal of online-first entries in ` journal_timeliness_checker` and entirely delete the entries from the DB after the timespan given there. Thus, they are either reloaded as valid articles or reinserted again if still present (r.g. in and RSS Feed). On the long run we avoid littering the DB with stale OF entries that never make it to an issue.

Reply:
journal_timeliness_checker is deprecated. The plan was to replace this program by pipelines/collect_journal_stats.cc together with logic inside of Zeder.

For transparency to the end user, it should be discussed with the librarians whether they want this information to stay in the database or not. Even for temporary listed information, there should be a change to zotero_cgi as well. If we decide to hide it from the frontend, there should at least be a comment there (and maybe in the wiki too).

Reply:
As discussed: For the moment `journal_timeliness_checker` is still in use. Online first now shows up in the status with respective comments.

Reply:
I wonder if the logic in Zeder is implemented in the meantime... maybe we could rename journal_timeliness_checker and remove some deprecated functionality when time permits.